### ISSUE TITLE
I think that computation of fast sparse cosine similarity fails if one vector is empty

### ISSUE THREAD

**Issue by searchivarius** [searchivarius/nmslib#72]
Likely the result is NAN

**Comment by goerch** [comment]
I seem to be running just now into this problem. The reduced test case

----------
import nmslib
import numpy as np
import sklearn
from scipy.sparse import csr_matrix
from scipy.stats import mode
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split

print(sklearn.__version__)
print(nmslib.__version__)

X, y = make_classification()
X_train, X_test, y_train, y_test = train_test_split(X, y)
index = nmslib.init(method='hnsw', space='cosinesimil_sparse', data_type=nmslib.DataType.SPARSE_VECTOR)
index.addDataPointBatch(csr_matrix(X_train))
index.createIndex()

neighbors = index.knnQueryBatch(csr_matrix(X_test), k=1)
indices, _ = zip(*neighbors)
votes, _ = mode(np.array(y_train)[np.vstack(indices)], axis=1)
votes.flatten()

neighbors = index.knnQueryBatch(csr_matrix([]), k=1)
indices, _ = zip(*neighbors)
votes, _ = mode(np.array(y_train)[np.vstack(indices)], axis=1)
votes.flatten()
----------

produces the trace back

----------
0.23.1
2.0.6
Check failed: obj2->datalength() > 0
Traceback (most recent call last):
[...]
    neighbors = index.knnQueryBatch(csr_matrix([]), k=1)
RuntimeError: Check failed: it's either a bug or inconsistent data!
----------

**Comment by searchivarius** [comment]
@goerch right, this actually makes sense now. This needs to be fixed. As a temporary workaround, empty vectors can be replaced with some dummy vectors having zero value of the zero coordinate.

**Comment by valentas-kurauskas** [comment]
Has this been fixed? I can't reproduce the error.
